cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/ConcurrentStreamWriter.java

```diff
@@ -111,6 +111,11 @@
     this.createLock = new ReentrantLock();
   }
 
+  public void close(Id.Stream streamId) throws IOException, NotFoundException {
+    EventQueue eventQueue = getEventQueue(streamId);
```

Comment:
For closing purpose, probably you don't want to create a new `EventQueue` if it doesn't exists. So, instead of calling `getEventQueue`, better do 

```java
  Closeables.close(eventQueues.get(streamId), false);
```